scummvm/scummvm
# SCUMM: Fix the Nazis' uniforms in the corridors of Castle Brunwald (FM-TOWNS)

*This is a small, non-critical oversight from the original game. This PR can be put on hold for the 2.6.0 freeze.* 

## Context

The 256-color versions of Indy3 are "interesting" in that they have various color problems… sometimes the FM-TOWNS version fixed some of them while the PC VGA version didn't, or vice versa. (And some problems were never fixed anywhere.)

Anyway, I don't have the FM-TOWNS version (crazy crazy prices) but ATMachine had a very interesting analysis:
https://web.archive.org/web/20111015233333/http://home.comcast.net/~ervind/ij3misc.html

One particular point struck me: *"Also in the FM Towns version, a similar error happens with the Nazis in the Castle Brunwald corridors: the game scripts change their green uniforms to grey."*

The problem is that it is a continuity error, then:
https://www.youtube.com/watch?v=ahOrPF1PjcY&t=200s  
https://web.archive.org/web/20131210202142im_/http://i20.photobucket.com/albums/b241/ATMachine/lecmisc/ij3towns7.png

In the corridors, the guards (and Biff) wear gray, but if you fight them one second later they wear green. They're also always in green when you are in the zeppelin maze. So yes, indeed, it looks like the FM-TOWNS porters forgot to turn off the palette override for these costumes in the corridor view.

So, if there's any `SO_PALETTE` override given to `ActorOps()` for these costumes in the corridor rooms, the following PR will just ignore it, since it was just meant for the EGA version, and the PC VGA version doesn't have it anymore.

But… I don't have the FM-TOWNS version of Indy 3, so I couldn't test this! I just made a guess and tried to emulate this condition by tweaking my PC VGA version.

So it'd be better **not to merge this** until someone with an FM-TOWNS Indy3 can tell if this works ⚠️. Thanks in advance!

## How to test

1. Launch the FM-TOWNS version of Indy 3
2. Explore all the corridors of Castle Brunwald. All the Nazi guards (and Biff) in the corridor view (= top-down perspective) should wear green as they do when you meet them. (Higher officials are not supposed to wear green, as far as I know.)
